RecoEgamma/EgammaElectronProducers/plugins/ElectronSeedConverter.cc

```diff
@@ -0,0 +1,105 @@
+#include "DataFormats/EgammaReco/interface/ElectronSeedFwd.h"
+
+
+class ElectronSeedConverter : public edm::global::EDProducer<> {
+public:
+  explicit ElectronSeedConverter(const edm::ParameterSet&);
+
+  void produce(edm::StreamID, edm::Event&, const edm::EventSetup&) const final;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+    const edm::EDGetTokenT<TrajectorySeedCollection> initialSeedsToken_;
+    const edm::EDGetTokenT<reco::EleSeedHostCollection> matchedEleSeedSoAToken_; 
+    const edm::EDPutTokenT<reco::ElectronSeedCollection> putToken_;
+	edm::EDGetTokenT<std::vector<reco::SuperClusterRef>> superClustersTokens_;
+};
+
+
+void ElectronSeedConverter::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+    edm::ParameterSetDescription desc;
+    desc.add<edm::InputTag>("initialSeeds", {"hltElePixelSeedsCombined"});
+    desc.add<edm::InputTag>("eleSeedsSoA", edm::InputTag{"ElectronNHitSeedAlpakaProducer"});
+  	desc.add<edm::InputTag>("superClusters", {"hltEgammaSuperClustersToPixelMatch"});
+    descriptions.add("electronNHitSeedConverter", desc);
+}
+
+ElectronSeedConverter::ElectronSeedConverter(const edm::ParameterSet& pset)
+    :initialSeedsToken_(consumes(pset.getParameter<edm::InputTag>("initialSeeds"))),
+    matchedEleSeedSoAToken_(consumes<reco::EleSeedHostCollection>(pset.getParameter<edm::InputTag>("eleSeedsSoA"))),    
+    putToken_{produces<reco::ElectronSeedCollection>()} {
+	    superClustersTokens_ = consumes(pset.getParameter<edm::InputTag>("superClusters"));
+    }
+
+void ElectronSeedConverter::produce(edm::StreamID, edm::Event& event, const edm::EventSetup& iSetup) const {
+
+    auto const& view = event.get(matchedEleSeedSoAToken_).const_view();
+
+    reco::ElectronSeedCollection eleSeeds{};
+    // auto eleSeeds = std::make_unique<reco::ElectronSeedCollection>();
+    // eleSeeds->reserve(eleSeedSoAView.metadata().size());
+
+    std::map<int, reco::SuperClusterRef> superClusterRefMap_;
+    std::map<int, TrajectorySeed> seedRefMap_;
+
+    int i=0;
+	for (auto& superClusRef : event.get(superClustersTokens_)) {
+		superClusterRefMap_[i] = superClusRef;
+		++i;
+    }
+
+    i=0;
+    for (auto& initialSeedRef : event.get(initialSeedsToken_)) {
+        seedRefMap_[i] = initialSeedRef;  
+        ++i;
+    }
+
+    for (int i = 0; i < view.metadata().size(); ++i) {
+        if (view[i].isMatched() > 0) {
+            int matchedScID = view[i].matchedScID();
+            auto scIter = superClusterRefMap_.find(matchedScID);
+            std::cout << "  matchedScID: " << view[i].matchedScID() << std::endl;
+
+            if (scIter != superClusterRefMap_.end()) {
+                const reco::SuperClusterRef& superClusRef = scIter->second;
+                auto seedIter = seedRefMap_.find(view[i].id());
+                if (seedIter != seedRefMap_.end()) {
+                    const TrajectorySeed& matchedSeed = seedIter->second;
```

Comment:
With `seedRefMap` being a `vector`, this piece would be written as
```suggestion
                auto seedIter = seedRefMap.find();
                if (view[i].id() < seedRefMap.size) {
                    const TrajectorySeed& matchedSeed = seedRefMap_[view[i].id()];
```